src/components/operations/operation-details/react/runtime/operation-console/ws-utilities/websocketClient.ts

```diff
@@ -0,0 +1,133 @@
+export enum LogItemType {
+    Connection = "Connection",
+    SendData = "SendData",
+    GetData = "GetData",
+    Error = "Error"
+}
+
+export interface LogItem {
+    logTime: string;
+    logData: string;
+    logType: LogItemType;
+}
+
+export class WebsocketClient {
+    private websocket: WebSocket;
+    private logs: LogItem[];
+
+    constructor(private websocketUrl?: string) {
+        this.onOpenConnection = this.onOpenConnection.bind(this);
+        this.onCloseConnection = this.onCloseConnection.bind(this);
+        this.onErrorConnection = this.onErrorConnection.bind(this);
+        this.onMessageConnection = this.onMessageConnection.bind(this);
+        this.logs = [];
+    }
+
+    public isConnected: boolean;
+
+    public onOpen: (message: string) => void;
+    public onClose: (event: CloseEvent) => void;
+    public onError: (error: string) => void;
+    public onMessage: (data: MessageEvent<any>) => void;
+    public onLogItem: (data: LogItem) => void;
+
+    public get logItems() {
+        return this.logs;
+    }
+
+    public connect(websocketUrl?: string, subProtocol?: string) {
+        this.websocketUrl = websocketUrl || this.websocketUrl;
+        this.logDataItem(`Connecting to ${this.websocketUrl}`, LogItemType.Connection);
+
+        this.websocket = new WebSocket(this.websocketUrl, subProtocol);
+        this.websocket.onopen = this.onOpenConnection;
+        this.websocket.onclose = this.onCloseConnection;
+        this.websocket.onerror = this.onErrorConnection;
+        this.websocket.onmessage = this.onMessageConnection;
+    }
```

Comment:
I wonder if we don't need to dispose the WebScoket / unsubscribe to events when the comments i getting unmounted.